synapse/http/server.py

```diff
@@ -393,7 +393,7 @@ def respond_with_json_bytes(request, code, json_bytes, send_cors=False,
             http://www.w3.org/TR/cors/
     Returns:
         twisted.web.server.NOT_DONE_YET"""
-
+    json_bytes = json_bytes.encode('utf8')
```

Comment:
`.encode` goes from `str` to `bytes`, and according to the comment at line 391 (and the name of the variable), json_bytes is already a `bytes`, so this doesn't look right.